### ISSUE TITLE
MSS510x Firmware update to 3.2.2

### ISSUE THREAD

**Issue by chrisrpi** [krahabb/meross_lan#89]
<!-- Before you open a new issue, search through the existing issues to see if others have had the same problem.

Hi...fab component...many thanks!

I have just upgraded firmware my Meross MSS510x switch (seemed like the right thing to do at the time!) and it seems to have broken the connection with Home Assistant through this component. I tried restarting the switch, HA and the integration, but keep getting 'unable to connect'. I read on Giters that entering a dummy device key and reclicking a few times might work, but I can't get past 'unable to connect'. Had a look at the supervisor logs etc but couldn't see anything related. Happy to provide any info that might help, or implement a work round etc.

Many thanks,

Chris.

**Comment by chrisrpi** [comment]
Hi...just updated the firmware, which seems to have broken the integration with HA. Tried restarting HA, integration and he switch but still get 'unable to connect'. Read on Giters that entering a dummy key and rechecking a few times might work, but I can't get past 'unable to connect'. Couldn't see awnythgin relevant in Supervisor log, but happy to provide and info that might help or to use a workaround etc.

Many thanks for the great integration!

**Comment by krahabb** [comment]
Hello @chrisrpi ,
By the look of it, it seems you need the original device key which you can recover in meross_lan configuration by entering your Meross credentials when prompted.
The flow is still a bit 'rusty' but you should be able to get to the login prompt and once recovered have the device key filled with the correct one.
If you folllow a bit on #69, you should be able to get to the point
Basically, in order for meross_lan to ask you the credentials you have to get an 'invalid key' error when configuring the device.

**Comment by chrisrpi** [comment]
Thanks so much for your response … an update for your integration came through today … I updated and retried adding the switch. It asked for my meross login credentials straight away … switch added!

Thank you so much, you are one of the far greater minds than mine making life easier for others!

**Comment by krahabb** [comment]
Thank you..that' too much ;)
